R/pkg/R/utils.R

```diff
@@ -339,20 +339,40 @@ varargsToEnv <- function(...) {
 varargsToStrEnv <- function(...) {
   pairs <- list(...)
   env <- new.env()
-  for (name in names(pairs)) {
-    value <- pairs[[name]]
-    if (!(is.logical(value) || is.numeric(value) || is.character(value) || is.null(value))) {
-      stop(paste0("Unsupported type for ", name, " : ", class(value),
-           ". Supported types are logical, numeric, character and NULL."))
-    }
-    if (is.logical(value)) {
-      env[[name]] <- tolower(as.character(value))
-    } else if (is.null(value)) {
-      env[[name]] <- value
-    } else {
-      env[[name]] <- as.character(value)
+  nameList <- names(pairs)
+  ignoredNames <- list()
+  i <- 1
```

Comment:
why is `i` needed here?
could you elaborate on changes in this file? why do we need them?